vault/core.go

```diff
@@ -1339,10 +1336,9 @@ func (c *Core) Shutdown() error {
 	c.stateLock.Lock()
 	defer c.stateLock.Unlock()
 
-	doneCh := c.shutdownDoneCh.Load().(chan struct{})
-	if doneCh != nil {
-		close(doneCh)
-		c.shutdownDoneCh.Store((chan struct{})(nil))
+	if !c.shutdownDoneChClosed {
```

Comment:
Isn't this racy? What if two calls to Shutdown are made at the same time?  Then we could close twice and panic.

Reply:
This is protected by the `stateLock` mutex above, so the second call to shutdown would always find `shutdownDoneChClosed == true`

Reply:
True!